### DISCUSSION TITLE
Not finding how to login with SSO

### DISCUSSION THREAD

**Discussion by nvds71** [CrowdStrike/falconpy#1260] [Q&A] [answered]
**Documentation page topic**
No functionality found or example code showing how to login using SSO with Falconpy.

**Describe the error or typo**
We authenticate to Falcon using SSO (microsoftonline), which doesn't require a password. However, there seems to be no functionality available to login using this method with Falconpy. All examples use a client secret, which we obviously don't have.
Any suggestions what I might be doing wrong, or has this just not been implemented in falconpy?

**Documentation repository:**
 - [ ] GitHub Wiki
 - [ ] falconpy.io
 - [ ] Source code - docstring

**Comment by jshcodes** [comment, marked as answer]
Hi @nvds71 -

When you login to Falcon using SSO are you able to access the API clients and keys page? You can generate your API client ID and secret here, and then assign the necessary scopes.  (Using the navigation: `Support and Resources` -> `API clients and keys`)

FalconPy does not natively support SSO authentication to a Falcon tenant, but this _can_ be accomplished with additional coding. 

Thank you for the question! 😄

**Reply by nvds71** [reply]
Thank you very much @jshcodes! I created an API client and key and added that to my code. This worked immediately.
